components/camel-git/src/main/java/org/apache/camel/component/RepositoryFactory.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.camel.component.git.GitEndpoint;
+import org.apache.camel.util.ObjectHelper;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.eclipse.jgit.util.SystemReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class RepositoryFactory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RepositoryFactory.class);
+    private static final SystemReader DEFAULT_INSTANCE;
+
+    static {
+        DEFAULT_INSTANCE = SystemReader.getInstance();
+    }
+
+    private RepositoryFactory() {
+    }
+
+    public static Repository of(GitEndpoint endpoint) throws IOException {
+        if (ObjectHelper.isNotEmpty(endpoint.getGitConfigFile())) {
+            return fromPathToConfigFile(endpoint, endpoint.getGitConfigFile());
+        }
+        return getRepository(endpoint, DEFAULT_INSTANCE);
+    }
+
+    private static Repository fromPathToConfigFile(GitEndpoint endpoint, String pathToConfigFile) throws IOException {
+        if (ObjectHelper.isEmpty(pathToConfigFile)) {
+            throw new IllegalArgumentException("Path to git config file must be supplied");
+        }
+        File gitConfigFile;
+        if (pathToConfigFile.startsWith("/")) { //load from system
```

Comment:
Wouldn't this break on Windows?

Reply:
Sure, this will. I will change.

Reply:
@oscerd 
> Maybe we can also point a remote configfile through resource helper and copy locally in the git repository before using it.

Do you think We do it using same config param (`gitConfigFile`) or other? like `remoteGitConfigFile`

Reply:
Probably with remoteGitConfigFile. With resource helper you could able to grab file content directly from classpath, filesystem, http etc.

Reply:
Done.